### PR TITLE
fix 'importsNotUsedAsValues' deprecation error

### PR DESCRIPTION
**Pull request by TBonnin**
## Describe your changes
We are running into the following error when compiling individual package compare to using `npm run ts-build`:
> Option 'importsNotUsedAsValues' is deprecated and will stop functioning in TypeScript 5.5. Specify compilerOption '"ignoreDeprecations": "5.0"' to silence this error

This seems to come from the fact that different packages uses diferrent version of typescript.
This PR aligns typescript version for all packages in the npm workspace, fixing errors caught by the compiler and ignore v5 warnings so all packages can be compiled individually or as a workspace

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

**Review by khaliqgant — COMMENTED** [review]
Does this impact how the build process works when running the nango cli and the compilation there?

**Comment by TBonnin** [comment]
> Does this impact how the build process works when running the nango cli and the compilation there?

do you mean when  running `npm run cli:watch`? I tried it and it compile without any issue. Same for `dw` or `dwa`

File: packages/shared/lib/services/sync/config/config.service.ts [hunk 1]
```diff
@@ -635,8 +635,8 @@ export async function getNangoConfigIdAndLocationFromId(id: number): Promise<{ n
         .from<SyncConfig>(TABLE)
         .select(`${TABLE}.nango_config_id`, `${TABLE}.file_location`)
         .where({
-            [`${TABLE}.id`]: id,
-            [`${TABLE}.deleted`]: false
+            id,
+            deleted: false
```
Comment (khaliqgant) [inline packages/shared/lib/services/sync/config/config.service.ts:639]: Was this throwing an error? Otherwise this change seems unrelated

File: packages/shared/lib/services/sync/data/records.service.ts [hunk 2]
```diff
@@ -467,7 +467,7 @@ export async function getAllDataRecords(
 
         const customerResult = result.map((item) => item.record);
 
-        if (customerResult.length > (limit || 100)) {
+        if (customerResult.length > +(limit || 100)) {
```
Comment (khaliqgant) [inline packages/shared/lib/services/sync/data/records.service.ts]: Would rather Number to cast than + as I think it is more readable

**Comment by khaliqgant** [comment]
> > Does this impact how the build process works when running the nango cli and the compilation there?
> 
> do you mean when running `npm run cli:watch`? I tried it and it compile without any issue. Same for `dw` or `dwa`

My question was regarding actually compilation of sync scripts using the nango cli and making sure that works as expected and that this scenario was tested

File: packages/shared/lib/services/sync/config/config.service.ts (hunk 1 above)
Comment (TBonnin) [reply packages/shared/lib/services/sync/config/config.service.ts:639]: it is throwing an error: it doesn't like prefixed column like `_nango_sync_configs.id` unless there is a join

**Comment by khaliqgant** [comment]
> > > Does this impact how the build process works when running the nango cli and the compilation there?
> > 
> > 
> > do you mean when running `npm run cli:watch`? I tried it and it compile without any issue. Same for `dw` or `dwa`
> 
> My question was regarding actually compilation of sync scripts using the nango cli and making sure that works as expected and that this scenario was tested

I tested this locally and saw no issues.

**Review by khaliqgant — APPROVED** [review]
Non blocking `Number` vs usage of `+` feedback inline

**Comment by TBonnin** [comment]
> > > > Does this impact how the build process works when running the nango cli and the compilation there?
> > > 
> > > 
> > > do you mean when running `npm run cli:watch`? I tried it and it compile without any issue. Same for `dw` or `dwa`
> > 
> > 
> > My question was regarding actually compilation of sync scripts using the nango cli and making sure that works as expected and that this scenario was tested
> 
> I tested this locally and saw no issues.

Same I have compiled a couple of test sync and the compiled code happened to be identical. (same md5 hash once removing the sourcemap)
